homeassistant/components/qnap/sensor.py

```diff
@@ -56,31 +57,46 @@
 NOTIFICATION_TITLE = "QNAP Sensor Setup"
 
 _SYSTEM_MON_COND = {
-    "status": ["Status", None, "mdi:checkbox-marked-circle-outline"],
-    "system_temp": ["System Temperature", TEMP_CELSIUS, "mdi:thermometer"],
+    "status": ["Status", None, "mdi:checkbox-marked-circle-outline", None],
+    "system_temp": ["System Temperature", TEMP_CELSIUS, None, DEVICE_CLASS_TEMPERATURE],
 }
 _CPU_MON_COND = {
-    "cpu_temp": ["CPU Temperature", TEMP_CELSIUS, "mdi:thermometer"],
-    "cpu_usage": ["CPU Usage", PERCENTAGE, "mdi:chip"],
+    "cpu_temp": ["CPU Temperature", TEMP_CELSIUS, None, DEVICE_CLASS_TEMPERATURE],
+    "cpu_usage": ["CPU Usage", PERCENTAGE, "mdi:chip", None],
 }
 _MEMORY_MON_COND = {
-    "memory_free": ["Memory Available", DATA_GIBIBYTES, "mdi:memory"],
-    "memory_used": ["Memory Used", DATA_GIBIBYTES, "mdi:memory"],
-    "memory_percent_used": ["Memory Usage", PERCENTAGE, "mdi:memory"],
+    "memory_free": ["Memory Available", DATA_GIBIBYTES, "mdi:memory", None],
+    "memory_used": ["Memory Used", DATA_GIBIBYTES, "mdi:memory", None],
+    "memory_percent_used": ["Memory Usage", PERCENTAGE, "mdi:memory", None],
 }
 _NETWORK_MON_COND = {
-    "network_link_status": ["Network Link", None, "mdi:checkbox-marked-circle-outline"],
-    "network_tx": ["Network Up", DATA_RATE_MEBIBYTES_PER_SECOND, "mdi:upload"],
-    "network_rx": ["Network Down", DATA_RATE_MEBIBYTES_PER_SECOND, "mdi:download"],
+    "network_link_status": [
+        "Network Link",
+        None,
+        "mdi:checkbox-marked-circle-outline",
+        None,
+    ],
+    "network_tx": ["Network Up", DATA_RATE_MEBIBYTES_PER_SECOND, "mdi:upload", None],
+    "network_rx": [
+        "Network Down",
+        DATA_RATE_MEBIBYTES_PER_SECOND,
+        "mdi:download",
+        None,
+    ],
 }
 _DRIVE_MON_COND = {
-    "drive_smart_status": ["SMART Status", None, "mdi:checkbox-marked-circle-outline"],
-    "drive_temp": ["Temperature", TEMP_CELSIUS, "mdi:thermometer"],
+    "drive_smart_status": [
+        "SMART Status",
+        None,
+        "mdi:checkbox-marked-circle-outline",
+        None,
+    ],
+    "drive_temp": ["Temperature", TEMP_CELSIUS, None, DEVICE_CLASS_TEMPERATURE, None],
```

Comment:
Device class & icon swapped?

```suggestion
    "drive_temp": ["Temperature", TEMP_CELSIUS, None, None, DEVICE_CLASS_TEMPERATURE],
```

Reply:
Fixed in e05330a